src/backend/gporca/libgpopt/src/base/CCostContext.cpp

```diff
@@ -437,6 +437,61 @@ CCostContext::FBetterThan(const CCostContext *pcc) const
 		}
 	}
 
+	// if multistageAgg flag is true and the distribution type is not replicated/universal
+	// mark multistage agg plan is better and should override cost based check
+	// between multistage agg and single stage agg. When data distribution is universal or replicated,
+	// it means that the same data is available across all the nodes, there is no need for motions
+	// between nodes during aggregation. What if we force a multi-stage plan where no motion exists,
+	// we would get 2 consecutive aggs that are redundant. So multi-stage aggregation
+	// may not be necessary in this scenario, instead single-stage aggregation approach can be used,
+	// where each node independently aggregates its local data and the final result is obtained
+	// by combining the local aggregates.
+	if (GPOS_FTRACE(EopttraceForceMultiStageAgg))
+	{
+		if ((IsMultiStageAggCostCtxt(pcc) && IsSingleStageAggCostCtxt(this)))
+		{
+			if (CDistributionSpec::EdtStrictReplicated ==
+					Pdpplan()->Pds()->Edt() ||
+				CDistributionSpec::EdtTaintedReplicated ==
+					Pdpplan()->Pds()->Edt() ||
+				CDistributionSpec::EdtUniversal == Pdpplan()->Pds()->Edt())
```

Comment:
```suggestion
	switch (Pdpplan()->Pds()->Edt())
	{
	    case CDistributionSpec::EdtStrictReplicated:
	    case CDistributionSpec::EdtTaintedReplicated:
	    case CDistributionSpec::EdtUniversal:
	        return true;
	    default:
	        return false;
	}
```

Can we use switch case?

Reply:
yes we can use here . I have updated. Thanks  :)